### DISCUSSION TITLE
Does unikraft support Raspberry or any other embedded system?

### DISCUSSION THREAD

**Discussion by sarthak** [unikraft/unikraft#311] [Q&A]
Hello,

Our research group is interested in evaluating feasibility of unikernels in embedded systems. [We saw a talk by Felipe Huici](https://connect.linaro.org/resources/lvc20/lvc20-107/) in which he has asserted that Unikraft has been ported to Raspberry Pi, however, in the build system I do not see any target for it? In particular in the slides I see that `libraspi` is used as a platform library so I suppose this was developed as an external platform library for unikraft, however, in the unikraft github organization this library is not available. Is it not open sourced or am I missing something? Can someone guide me what configuration options need to be selected to build a baremetal target for Raspberry Pi? It is fine even if Unikraft doesn't support Raspberry Pi but supports any other embedded system bare metal. Thank you.

**Comment by cristian-vijelie** [comment]
At this momment, Unikraft doesn't support any baremetal platforms. All the work that has been done is private, for RaspberryPi and xilinx-zynqmp. Currently, no one that I know of works on baremetal support anymore. If you or your group want to contribute to this, any help is appreciated.

Sorry for the (very) late answer.
